cli/report/report.py

```diff
@@ -41,11 +54,30 @@ def __init__(self, firewatch_config: Configuration, job: Job) -> None:
             job (Job): A valid Job object representing the prow job being checked for failures/reported on.
         """
         self.logger = get_logger(__name__)
+        self._bugs_filed: list[str] = []
+        self._duplicate_bugs_commented: list[str] = []
+        self._rule_failure_pairs: list[dict[Any, Any]] = []
+        self._job: Job = job
+        self._gitleaks_detections: GitleaksDetectionCollection | None = None
 
         # If job is a rehearsal, exit 0
         if job.is_rehearsal:
             exit(0)
 
+        if gitleaks:
+            self.gitleaks_config = self._init_gitleaks_config()
+            self.gitleaks_config.start_detect_scan(keep_job_dir=True)
+            self._gitleaks_detections = self._find_gitleaks_detections()
+            if self._gitleaks_detections:
```

Comment:
```suggestion
            if self._gitleaks_detections := self._find_gitleaks_detections():
```

Reply:
I like this change and originally tried to use the `:=` operator here, but I receive this error when using Python 3.12.1.:

```
E       if self._gitleaks_detections := self._find_gitleaks_detections():
E          ^^^^^^^^^^^^^^^^^^^^^^^^^
E   SyntaxError: cannot use assignment expressions with attribute
```

It does not seem to like using `self` in this way, at least in the `__init__` method.